Updated error handling in ifw_c_lib

andrew-platt opened the pull request:
I updated the error handling and a few other things for the InflowWind c-bindings library interface.

Also rearranged the source code, and moved the regression tests to where they belong.

A few other details:

- Copyright is NREL since it was written while employed at NREL
- added `try` `except` logic as needed
- removed compile instructions (those are in readthedocs)
- removed input files from the source directories -- these go in r-test
- setup python regression testing and InflowWind_Driver regression test (same test, but python to ifw_c vs. fortran).  results are nearly identical
- cleaned up notes in the code and adjusted formatting as needed
- fixed a few missing error handling pieces (Calc_C and End_C only passed back single character of start of message)
- other stuff that I don't remember
- Updated CMake
- Added VS Solution
- Removed `archive` directory -- we don't keep such things (that's what Git history is for)

andrew-platt committed:
ifw_c_lib:  rearrange repo, update cmakelists

andrew-platt committed:
Merge remote-tracking branch 'OpenFAST/dev' into f/CCT2-IFW_wrapper

andrew-platt committed:
ifw_c_lib: move test case to r-test, rearrange

andrew-platt committed:
ifw_c_lib: modify error handling

andrew-platt committed:
ifw_c_lib: add ifw driver test case infrastructure

Commit:
IfW_c_lib: add VS project and solution

andrew-platt committed:
IfW_c_lib: minor notes in r-test

Need to set this aside for a bit, so adding temporary notes for later

andrew-platt committed:
IfW: redundancy in error handling of InflowWind_ParseInputFileInfo routine

andrew-platt committed:
NWTC lib: add InitFileInfo_FromNullCString subroutine

This routine allows for passing an input file as one long string with C_NULL_CHAR deliations between lines of the input file
Set interface for InitFileInfo.

andrew-platt committed:
IfW: Change input file string passing for ifw_c_lib

 - cherry-pick of commit for updates to NWTC library
 - Update the strings passed to the python lib into an array of variable length strings
 - Change the copyright (this was developed at NREL, so NREL holds the copyright)
 - Change the string passing of the input files
   -- now the driver passes an array of variable strength strings to the python lib
   -- the python lib convers this to a C_NULL_CHAR delimited single string that is passed to IfW_c_lib
   -- the ifw_c_lib now picks apart this single string into different input file lines based on the C_NULL_CHAR locations

andrew-platt committed:
NWTC lib: add unit test for InitFileInfo_FromNullCString subroutine and fix broken test

Also reorder part of the InitFileInfo_FromStringArray routine

andrew-platt committed:
ifw_c_lib: improve error handling and add results

andrew-platt committed:
ifw_c_lib: configure inflowwind python driver regression test

andrew-platt committed:
Merge remote-tracking branch 'nrmendoza/f/CCT2-IFW_wrapper' into f/CCT2-IFW_wrapper

rafmudaf commented on modules/inflowwind/CMakeLists.txt:
```diff
@@ -51,17 +51,8 @@ set(IFW_SOURCES
-# set_target_properties(ifw_c_lib PROPERTIES PREFIX "" SUFFIX ".dll")
-
+# c-bindings interface library
+add_library(ifw_c_lib SHARED src/IfW_C.f90)
```
@andrew-platt what do you think about adding a CMake flag to compile all C-bound interfaces instead of compiling them by default?

andrew-platt replied:
I like that idea.  We'll need to tie those optional targets to the respective regression tests as we add them.

rafmudaf reviewed (approved):
Very nicely done @andrew-platt!

rafmudaf commented:
I just noticed this in the description
```
Added VS Solution
```
but I dont see it in the files changes.

andrew-platt commented:
Hmm.... Maybe I forgot to commit the VS solution.  I'll check the other computer and see if I have it there uncommitted...

edit:  yup.  Left that commit on the Windows VM without pushing it.

andrew-platt committed:
ifw_c_lib: error handling to prevent string overflow of ErrMsg_C

andrew-platt committed:
ifw_c_lib: update r-test pointer

Somehow we don't have the correct pointer to r-test with the f/weis updates for EllipticalWing

andrew-platt committed:
ifw_c_lib: removed unnecessary comments and variable declerations

andrew-platt committed:
ifw_c_lib: simplify error passing routine in the ifw_c.f90
